src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -194,7 +194,12 @@ export class TerminalCompletionService extends Disposable implements ITerminalCo
 			if (completions.resourceRequestConfig) {
 				const resourceCompletions = await this.resolveResources(completions.resourceRequestConfig, promptValue, cursorPosition, provider.id, capabilities, shellType);
 				if (resourceCompletions) {
-					completionItems.push(...resourceCompletions);
+					for (const item of resourceCompletions) {
+						const labels = completionItems.map(c => c.label);
+						if (!labels.includes(item.label)) {
+							completionItems.push(item);
+						}
+					}
 				}
```

Comment:
This is relatively expensive since you're doing a lot of duplicate work, try something like this:

```suggestion
					const labelSet = new Set(completionItems.map(c => c.label));
					for (const item of resourceCompletions) {
						if (!labels.has(item.label)) {
							completionItems.push(item);
						}
					}
				}
```